### PR TITLE
Allow ".edu.cn" accounts to have more time and disk quota

### PR DESCRIPTION
**Pull request by wwwjn**
### Reasons for making this change

Allow ".edu.cn" accounts to have more time and disk quota

### Related issues

None

### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [x] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [ ] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

File: codalab/model/bundle_model.py [hunk 1]
```diff
@@ -2481,12 +2481,12 @@ def add_user(
 
             time_quota = (
                 self.default_user_info['edu_time_quota']
-                if email.endswith(".edu")
+                if email.endswith(".edu") or email.endswith(".edu.cn")
```
Comment (percyliang) [inline codalab/model/bundle_model.py]: Can we pull this out into a separate function `is_academic_email` and generalize to `.edu.*`?

File: codalab/lib/spec_util.py [hunk 2]
```diff
@@ -106,3 +107,10 @@ def is_dashboard(name):
 
 def is_public_home(name):
     return name == 'home'
+
+
+def is_academic_email(email):
```
Comment (percyliang) [inline codalab/lib/spec_util.py]: This doesn't really belong in `spec_util`, which is the utility for specifications...you can just keep it in bundle_model for now.

Also, the logic we want is anything that ends with .edu or .edu.*, not just .edu.cn.

Comment (wwwjn) [reply codalab/lib/spec_util.py]: Thanks Percy! I'm really confused where to put this `is_academic_email ` function, and I will leave it in `bundle_model` for now.

File: codalab/model/bundle_model.py [hunk 3]
```diff
@@ -68,6 +68,7 @@
 
 SEARCH_KEYWORD_REGEX = re.compile('^([\.\w/]*)=(.*)$')
 SEARCH_RESULTS_LIMIT = 10
+EDU_USER_REGEXS = re.compile('@[\w\.-]+\.(edu|edu\.[a-z]{2})$')
```
Comment (epicfaace) [inline codalab/model/bundle_model.py]: change to REGEXES

File: codalab/model/bundle_model.py [hunk 4]
```diff
@@ -2475,18 +2476,27 @@ def add_user(
...
+            return len(email_suffix) > 0
+
         with self.engine.begin() as connection:
             now = datetime.datetime.utcnow()
             user_id = user_id or '0x%s' % uuid4().hex
 
             time_quota = (
                 self.default_user_info['edu_time_quota']
-                if email.endswith(".edu")
+                if is_academic_email(email)
```
Comment (epicfaace) [inline codalab/model/bundle_model.py:2495]: Can we add some unit tests for is_academic_email?

Comment (wwwjn) [reply codalab/model/bundle_model.py:2495]: Thanks Ashwin! For consistency, I add this unit test to bundle_model_test for now.

**Review by epicfaace — APPROVED** [review]

**Review by epicfaace — CHANGES_REQUESTED** [review]
We should also include ".ac.xx" domains -- see https://en.wikipedia.org/wiki/.ac_(second-level_domain)

**Comment by wwwjn** [comment]
> We should also include ".ac.xx" domains -- see https://en.wikipedia.org/wiki/.ac_(second-level_domain)

That's a good catch!

**Review by epicfaace — APPROVED** [review]

**Pull request merged**